src/tensor_ops/internal_reshapes.rs

```diff
@@ -0,0 +1,31 @@
+#[cfg(feature = "cuda")]
+use crate::prelude::Axes;
+#[cfg(feature = "cuda")]
+use std::vec::Vec;
+
+/// Moves all axes in Ax to the end of dims and strides and removes broadcasted dimensions
+/// so that a cuda kernel called for each physical element of the input tensor will place elements
+/// to be reduced with each other next to each other in memory.
```

Comment:
> will place elements to be reduced with each other next to each other in memory.

How exactly does this happen? Changing strides won't actually move stuff around in memory, so I guess i'm not seeing how we can depend on the sequential addressing used for the chunk_min/chunk_max/chunk_sum.

What am I missing?